### PR TITLE
Improving warning text for eth_sign

### PR DESCRIPTION
**Pull request by ryanml**
Implements the first two steps for: https://github.com/MetaMask/metamask-extension/issues/11337

Screen Shot 2021-09-03 at 11 22 51 AM

### CODE REVIEW

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -1958,7 +1958,7 @@
     "message": "Sign"
   },
   "signNotice": {
-    "message": "Signing this message can have \ndangerous side effects. Only sign messages from \nsites you fully trust with your entire account.\n This dangerous method will be removed in a future version. "
+    "message": "We are unable to estimate the result of this signature. This signature could potentially perform any operation on your account's behalf, including granting complete control of your account and all of its assets to the requesting site. Only sign this message if you know what you're doing or completely trust the requesting site."
```
Comment (adonesky1) [inline app/_locales/en/messages.json:1961]: minor proposed change to copy : `We are unable to **predict** the result of this signature...`

Comment (ryanml) [reply app/_locales/en/messages.json:1961]: ++

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by NiranjanaBinoy — APPROVED** [review]
lgtm

**Pull request merged**